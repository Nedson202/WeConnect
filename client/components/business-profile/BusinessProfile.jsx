import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import 'rc-pagination/assets/index.css';
import ImageZoom from 'react-medium-image-zoom';
import { Link } from 'react-router-dom';
import ReactStars from 'react-stars';
import PropTypes from 'prop-types';
import ReviewModal from './ReviewModal';
import ReviewList from './ReviewList';
import { reviewRequest, reviewUpdateRequest } from '../../actions/postReviewAction';
import { deleteBusiness, deleteReview } from '../../actions/deleteAction';
import { fetchReviews, fetchBusinessById } from '../../actions/fetchActions';
import BusinessImageUpload from '../ImageUpload/BusinessImageUpload';
import green from '../../images/default.jpeg';
import { businessImageUploader, uploadToCloudinary } from '../../actions/imageUpload';
import Spinner from '../common/Spinner';
import loader from '../../actions/loader';

let reviewToDelete;
/**
 * @class BusinessProfile
 *
 * @extends {Component}
 */
export class BusinessProfile extends Component {
  /**
   * @description Creates an instance of BusinessProfile.
   * @param {object} props
   * @memberof Profile
   */
  constructor(props) {
    super(props);

    this.state = {
      message: '',
      editMessage: '',
      image: '',
      rating: 0,
      editRating: 0,
      errors: {},
      current: 1,
      editStatus: null,
      isLoading: false
    };

    this.onBusinessDelete = this.onBusinessDelete.bind(this);
    this.onChange = this.onChange.bind(this);
    this.onImageChange = this.onImageChange.bind(this);
    this.onReviewPageChange = this.onReviewPageChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
    this.onImageSubmit = this.onImageSubmit.bind(this);
    this.onStarClick = this.onStarClick.bind(this);
    this.edit = this.edit.bind(this);
    this.cancelEdit = this.cancelEdit.bind(this);
    this.onReviewDelete = this.onReviewDelete.bind(this);
    this.deleteStatus = this.deleteStatus.bind(this);
    this.onReviewUpdate = this.onReviewUpdate.bind(this);
  }

  /**
   * @description Fetch reviews and business
   * @returns {undefined}
   * @memberof BusinessProfile
   */
  componentDidMount() {
    document.title = 'Business profile';
    const { id } = this.props.match.params;
    this.props.loader(true);
    this.props.fetchBusinessById(id);
    this.props.fetchReviews(id, 'page=1');
  }

  /**
   * @description Fetch reviews and business
   * @param{any} nextProps
   * @returns {undefined}
   * @memberof BusinessProfile
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.business) {
      const {
        name,
        email,
        address,
        location,
        category,
        description,
        image,
        averageRating
      } = nextProps.business;
      this.setState({
        name,
        email,
        address,
        location,
        category,
        image,
        description,
        averageRating,
      });
    }
  }
  /**
   * @description Handles image upload to firebase
   *
   * @param {any} event
   *
   * @returns {undefined}
   *
   * @memberof ImageUpload
   */
  onImageChange(event) {
    const image = event.target.files[0];
    const uploadPreset = process.env.UPLOAD_PRESET;
    const cloudinaryApi = process.env.CLOUDINARY_API;
    this.props.uploadToCloudinary(
      image,
      uploadPreset,
      cloudinaryApi,
    )
      .then(() => {
        this.setState({ image: this.props.image });
      });
  }

  /**
   * @description Fetch reviews and business
   *
   * @param {any} event
   *
   * @param {any} rating
   *
   * @returns {undefined}
   *
   * @memberof BusinessProfile
   */
  onChange(event) {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  }

  /**
   * @description Fetch reviews and business
   *
   * @param {any} newRating
   *
   * @returns {undefined}
   *
   * @memberof BusinessProfile
   */
  onStarClick(newRating) {
    if (!this.state.editStatus) {
      this.setState({ rating: newRating });
    }
    this.setState({ editRating: newRating });
  }

  /**
   * @description Fetch reviews and business
   *
   * @param {any} event
   *
   * @returns {undefined}
   *
   * @memberof BusinessProfile
   */
  onSubmit(event) {
    event.preventDefault();
    this.setState({ isLoading: true });
    const { params } = this.props;
    const { message, rating } = this.state;

    if (!message || rating === 0) {
      this.setState({ isLoading: false });
      return toastr.error('Please leave a review and rating');
    }

    this.props.reviewRequest(params.id, this.state, this.props.fetchBusinessById).then(() => {
      this.setState({ message: '', rating: 0, isLoading: false });
    });
  }

  /**
   * @description Deletes business from the database
   *
   * @param {any} event
   * @param {any} username
   *
   * @returns {undefined}
   *
   * @memberof BusinessProfile
   */
  onBusinessDelete(event) {
    event.preventDefault();
    const { history, params } = this.props;
    const { username } = this.props.user;
    this.props.deleteBusiness(params.id, username, history);
  }
  /**
   * @description Delete business
   *
   * @param {any} businessId
   *
   * @param {any} reviewToDelete
   *
   * @returns {undefined}
   *
   * @memberof ReviewList
   */
  onReviewDelete(businessId) {
    return () => {
      this.props.deleteReview(businessId, reviewToDelete, this.props.fetchBusinessById);
    };
  }
  /**
   * @description Handles image url submission to database
   *
   * @param {any} event
   *
   * @returns {undefined}
   *
   * @memberof ImageUpload
   */
  onImageSubmit(event) {
    event.preventDefault();
    this.props.businessImageUploader(this.props.params.id, this.state);
  }
  /**
   * @description Toggler for pagination
   *
   * @param {any} page
   *
   * @returns {undefined}
   *
   * @memberof BusinessList
   */
  onReviewPageChange(page) {
    const genPage = `page=${page}`;
    const { params } = this.props;

    this.setState({
      current: page,
    });

    this.props.fetchReviews(params.id, genPage);
  }
  /**
   * @description Fetch reviews and business
   * @param {any} reviewId
   * @returns {undefined}
   * @memberof EditReview
   */
  onReviewUpdate(reviewId) {
    this.setState({ isLoading: true });
    const {
      params
    } = this.props;
    const { state, cancelEdit } = this;
    const { editMessage, editRating } = this.state;
    if (!editMessage || editRating === 0) {
    this.setState({ isLoading: false });
    return toastr.error('Please leave a review and rating');
    }
    this.props.reviewUpdateRequest(
      params.id,
      reviewId,
      state,
      this.props.fetchBusinessById
    )
      .then(
        () => {
          cancelEdit()
          this.setState({ isLoading: false });
        });
  }
  /**
   * @description Edit business
   * @returns {undefined}
   * @param {any} reviewId
   * @memberof ReviewList
   */
  cancelEdit() {
    this.setState({ editStatus: null });
  }
  /**
   * @description Edit business
   * @returns {undefined}
   * @param {any} id
   * @memberof ReviewList
   */
  deleteStatus(id) {
    reviewToDelete = id;
  }
  /**
   * @description Edit business
   * @returns {undefined}
   * @param {any} review
   * @memberof ReviewList
   */
  edit(review) {
    const { id, message, rating } = review;
    return () => {
      this.setState({
        editStatus: id,
        editMessage: message,
        editRating: rating
      });
    };
  }
  /**
   * @description Deletes business from the database
   *
   * @param {any} event
   *
   * @returns {undefined}
   *
   * @memberof BusinessProfile
   */
  imageGenerator() {
    const { image } = this.state;
    if (!this.props.business && image) {
      return image;
    }
    if (image) {
      return image;
    }
    return green;
  }

  /**
   * @description Renders the component to the dom
   *
   * @returns {object} JSX object
   *
   * @memberof BusinessProfile
   */
  render() {
    const {
      reviews, business, params, user, paginate, isLoading
    } = this.props;

    const {
      name,
      address,
      location,
      email,
      description,
      category,
      averageRating,
    } = this.state;

    const {
      state,
      onStarClick,
      onChange,
      onSubmit,
      edit,
      cancelEdit,
      deleteStatus,
      onReviewDelete,
      onReviewPageChange,
      onReviewUpdate,
      onImageChange,
      onImageSubmit
    } = this;

    if (isLoading) {
      return <Spinner isLoading={isLoading} />;
    }

    const noReviews = (
      <h3>No reviews yet</h3>
    );

    const noBusiness = (
      <h3 className="text-center" id="no-business-found">No business match found</h3>
    );

    const deleteButton = (
      <button className="btn btn-danger" data-toggle="modal" data-target="#businessDeleteModal">
        <i className="fa fa-trash fa-lg" /> delete
      </button>
    );

    if (business.name === undefined) {
      return noBusiness;
    }

    return (
      <div>
        <div className="business-profile container">
          <div className="row">
            <div className="row col-sm-12 col-lg-10 offset-lg-1 profile-icon-style">
              <div className="image-button-holder col-lg-4">
                <ImageZoom
                  image={{
                      src: `${this.imageGenerator()}`,
                      alt: 'business image',
                      className: 'img',
                      style: { width: '320px', height: '300px' }
                    }}
                  zoomImage={{
                      src: `${this.imageGenerator()}`,
                      alt: 'business image',
                    }}
                />

                { !user ? null :
                <span>
                  { user.userId === (business && business.userId) ?
                    <button
                      className="btn image-edit-button"
                      data-toggle="modal"
                      data-target="#businessImageModal"
                    >
                      <i className="fa fa-edit fa-lg" />
                    </button> : null }
                </span> }
                <BusinessImageUpload
                  params={params}
                  onImageSubmit={onImageSubmit}
                  onImageChange={onImageChange}
                  uploadToCloudinary={this.props.uploadToCloudinary}
                  state={state}
                />
              </div>
              <div className="col-lg-6 business-details">
                <h4 className="business-name text-capitalize">{ !business ? name : business.name}</h4>
                <h4>Description:</h4>
                <p>{ !business ? description : business.description }</p>
                <p><i className="fa fa-envelope" /><span>{ !business ? email : business.email }</span></p>
                <p><i className="fa fa-map-marker" /><span>{ !business ? `${address} ${location}` : `${business.address} ${business.location}` }</span></p>
                <p><i className="fa fa-tags" /><span>{ !business ? category : business.category }</span></p>
                <ReactStars
                  count={5}
                  size={17}
                  edit={false}
                  color1="#333333"
                  color2="#ffaf00"
                  value={!business ? averageRating : business.averageRating}
                />
                { user.userId === (business && business.userId) ?
                  <ul className="list-unstyled list-inline pull-right" style={{ paddingTop: '30px' }}>
                    <li className="list-inline-item">
                      <Link className="btn edit-button" id="permission-button" to={`/updatebusiness/${params.id}`}>
                        <i className="fa fa-edit fa-lg" /> edit
                      </Link>
                    </li>
                    <li className="list-inline-item">
                      {deleteButton}
                    </li>
                  </ul> : null }
                { user.username === process.env.ADMIN ?
                  <ul
                    className="list-unstyled list-inline pull-right"
                    style={{ paddingTop: '30px', paddingBottom: '20px' }}
                  >
                    <li className="list-inline-item">
                      {deleteButton}
                    </li>
                  </ul> : null
                }
              </div>
            </div>
          </div>

          <div className="col-sm-12 col-lg-10 offset-lg-1">
            <ReviewModal
              onStarClick={onStarClick}
              onChange={onChange}
              onSubmit={onSubmit}
              state={state}
              user={user}
              business={business}
              isPosting={isLoading}
            />
          </div>

          <div className="col-sm-12 col-lg-10 offset-lg-1">
            { reviews.length === 0 ? noReviews : (
              <ReviewList
                reviews={reviews}
                deleteReview={this.props.deleteReview}
                fetchBusinessById={this.props.fetchBusinessById}
                fetchReviews={this.props.fetchReviews}
                reviewUpdateRequest={this.props.reviewUpdateRequest}
                paginate={paginate}
                params={params}
                user={user}
                isLoading={isLoading}
                onStarClick={onStarClick}
                onChange={onChange}
                onSubmit={onSubmit}
                state={state}
                edit={edit}
                cancelEdit={cancelEdit}
                deleteStatus={deleteStatus}
                onReviewDelete={onReviewDelete}
                onReviewPageChange={onReviewPageChange}
                onReviewUpdate={onReviewUpdate}
              />
                )}
          </div>
        </div>

        <div className="modal fade" id="businessDeleteModal" tabIndex="-1" role="dialog" aria-labelledby="confirmModalTitle" aria-hidden="true">
          <div className="modal-dialog modal-dialog-centered" role="document">
            <div className="modal-content">
              <div className="modal-body">
                <h4>Are you sure?</h4>
              </div>
              <div className="modal-footer">
                <button type="button" className="btn btn-outline-success" data-dismiss="modal">No</button>
                <button type="button" className="btn btn-outline-success" onClick={this.onBusinessDelete} data-dismiss="modal">Yes</button>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

BusinessProfile.propTypes = {
  fetchBusinessById: PropTypes.func.isRequired,
  reviewRequest: PropTypes.func.isRequired,
  reviewUpdateRequest: PropTypes.func.isRequired,
  // addFlashMessage: PropTypes.func.isRequired,
  deleteReview: PropTypes.func.isRequired,
  deleteBusiness: PropTypes.func.isRequired,
  fetchReviews: PropTypes.func.isRequired,
  businessImageUploader: PropTypes.func.isRequired,
  uploadToCloudinary: PropTypes.func.isRequired,
  loader: PropTypes.func.isRequired,
  paginate: PropTypes.object,
  isLoading: PropTypes.bool.isRequired,
  reviews: PropTypes.array,
  params: PropTypes.object.isRequired,
  business: PropTypes.object,
  user: PropTypes.object.isRequired,
  match: PropTypes.object.isRequired,
  image: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired,
};

BusinessProfile.contextTypes = {
  router: PropTypes.object.isRequired,
};

BusinessProfile.defaultProps = {
  reviews: [],
  business: {},
  paginate: {}
};

export const mapStateToProps = (state, props) => {
  const {
    businesses, isLoading, auth, image
  } = state;
  return {
    reviews: businesses.reviews,
    user: auth.user,
    business: businesses.business,
    params: props.match.params,
    paginate: businesses.paginationResult,
    image,
    isLoading,
  };
};

const mapDispatchToProps = dispatch => bindActionCreators({
  fetchBusinessById,
  reviewRequest,
  reviewUpdateRequest,
  deleteBusiness,
  fetchReviews,
  deleteReview,
  businessImageUploader,
  loader,
  uploadToCloudinary,
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(BusinessProfile);
