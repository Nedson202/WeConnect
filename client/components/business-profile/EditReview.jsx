import React from 'react';
import PropTypes from 'prop-types';
import ReactStars from 'react-stars';

/**
 * @class EditReview
 *
 * @extends {Component}
 */
const EditReview = ({
  onChange,
  onStarClick,
  state,
  onReviewUpdate,
  reviewId,
  cancelEdit
}) => {
  const { editMessage, editRating, isLoading } = state;
  return (
    <div className="row">
      <div className="col-lg-9">
        <button className="close" data-dismiss="modal" aria-label="Close" onClick={cancelEdit}>
          <span aria-hidden="true">&times;</span>
        </button>
        <div className="form-group">
          <textarea
            value={editMessage}
            onChange={onChange}
            type="text"
            name="editMessage"
            className="form-control edit-review-textarea"
            id="control-label"
            placeholder="update review"
            rows="3"
          />
        </div>
        <ReactStars
          count={5}
          onChange={onStarClick}
          size={17}
          color1="#333333"
          color2="#ffaf00"
          value={editRating}
        />
      </div>
      <div className="col-lg-3">
        <button
          disabled={isLoading}
          className="btn btn-outline-success update-review-button"
          id="update-review"
          onClick={() => onReviewUpdate(reviewId)}
        >
          { !isLoading ? <span>Update review</span> : <span>Updating <i className="fa fa-spinner fa-spin" /></span>}
        </button>
      </div>
    </div>
  );
};

EditReview.propTypes = {
  onReviewUpdate: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  onStarClick: PropTypes.func.isRequired,
  cancelEdit: PropTypes.func.isRequired,
  state: PropTypes.object.isRequired,
  reviewId: PropTypes.number.isRequired
};

export default EditReview;
