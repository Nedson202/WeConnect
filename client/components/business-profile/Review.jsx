import 'rc-pagination/assets/index.css';
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import ImageZoom from 'react-medium-image-zoom';
import ReactStars from 'react-stars';
import avatar from '../../images/user-avatar.png';
import EditReview from './EditReview';

/**
 * @class BusinessList
 * @extends {Component}
 */
class Review extends Component {
  /**
   * @description Display buttons
   * @returns {undefined}
   * @memberof ReviewList
   */
  displayButtons() {
    const {
      username,
      review,
      edit,
      deleteStatus,
    } = this.props;
    const { reviewer } = review;
    if (username === reviewer.username || username === process.env.ADMIN) {
      return (
        <div>
          <button
            className="close text-danger review-delete"
            data-toggle="modal"
            data-target="#confirmModal"
            onClick={() => deleteStatus(review.id)}
          >
            <i className="fa fa-trash" />
          </button>
          <button
            onClick={edit(review)}
            className="close"
          >
            <i className="fa fa-edit" />
          </button>
        </div>
      );
    }
    if (username !== reviewer.username && username !== process.env.ADMIN) {
      return null;
    }
    if (username === reviewer.username && username !== process.env.ADMIN) {
      return null;
    }
    if (username === reviewer.username || username === process.env.ADMIN) {
      return (
        <div>
          <button
            className="close text-danger"
            data-toggle="modal"
            data-target="#confirmModal"
          >
            <i className="fa fa-trash" />
          </button>
        </div>
      );
    }
  }

  /**
   * @description Renders the component to the dom
   *
   * @returns {object} JSX object
   *
   * @memberof ReviewList
   */
  render() {
    const {
      review,
      params,
      fetchReviews,
      reviewUpdateRequest,
      fetchBusinessById,
      cancelEdit,
      state,
      onReviewDelete,
      onChange,
      onStarClick,
      onReviewUpdate
    } = this.props;
    const {
      reviewer, message, rating, id, updatedAt
    } = review;

    const { image } = reviewer;

    if (state.editStatus === id) {
      return (
        <div>
          { !reviewer.image ? <img src={avatar} className="rounded-circle user-avatar" alt="Avatar" width="50" height="50" />
          : <img src={reviewer.image} className="rounded-circle user-avatar" alt="Avatar" width="50" height="50" />
        }
          <div className="review-body">
            <span className="textcase reviewer">{reviewer.username} </span>
            <EditReview
              reviewId={id}
              params={params}
              reviewUpdateRequest={reviewUpdateRequest}
              fetchReviews={fetchReviews}
              cancelEdit={cancelEdit}
              fetchBusinessById={fetchBusinessById}
              onChange={onChange}
              state={state}
              onStarClick={onStarClick}
              onReviewUpdate={onReviewUpdate}
            />
          </div>
        </div>
      );
    }

    return (
      <div>
        { !image ? <img src={avatar} className="rounded-circle user-avatar" alt="Avatar" width="50" height="50" />
          :
        <ImageZoom
          image={{
              src: `${image}`,
              alt: 'business image',
              className: 'img',
              style: {
 width: '50px', height: '50px', marginRight: '15px', float: 'left', borderRadius: '50%'
}
            }}
          zoomImage={{
              src: `${image}`,
              alt: 'business image',
            }}
        />
        }
        <div className="review-body">
          <span className="textcase reviewer">{reviewer.username} </span>
          {
            this.displayButtons()
          }
          <div id={id}>
            <p>{message}</p>
            <ReactStars
              count={5}
              edit={false}
              half
              size={17}
              color1="#333333"
              color2="#ffaf00"
              value={rating}
            />
          </div>
          <p
            className="mb-0 small font-weight-medium
            mb-1 text-muted lts-2px"
          >
            {moment(updatedAt).format('MMMM Do YYYY, h:mm:ss a')}
          </p>
        </div>

        <div className="modal fade" id="confirmModal" tabIndex="-1" role="dialog" aria-labelledby="confirmModalTitle" aria-hidden="true">
          <div className="modal-dialog modal-dialog-centered" role="document">
            <div className="modal-content">
              <div className="modal-body">
                <h4>Are you sure?</h4>
              </div>
              <div className="modal-footer">
                <button type="button" className="btn btn-outline-success" data-dismiss="modal">No</button>
                <button type="button" className="btn btn-outline-success" onClick={onReviewDelete(params.id, id)} data-dismiss="modal">Yes</button>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

Review.propTypes = {
  fetchBusinessById: PropTypes.func.isRequired,
  reviewUpdateRequest: PropTypes.func.isRequired,
  onReviewDelete: PropTypes.func.isRequired,
  onStarClick: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  onReviewUpdate: PropTypes.func.isRequired,
  deleteStatus: PropTypes.func.isRequired,
  edit: PropTypes.func.isRequired,
  cancelEdit: PropTypes.func.isRequired,
  fetchReviews: PropTypes.func.isRequired,
  review: PropTypes.object.isRequired,
  params: PropTypes.object.isRequired,
  state: PropTypes.object.isRequired,
  username: PropTypes.string
};

Review.defaultProps = {
  username: ''
};

export default Review;
