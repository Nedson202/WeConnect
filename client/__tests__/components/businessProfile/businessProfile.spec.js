import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';
import sinon from 'sinon';
import connectedBusinessProfile, { BusinessProfile } from '../../../components/business-profile/BusinessProfile';
import businessData from '../../__mockData__/business';
import userData from '../../__mockData__/userData';
import { mapStateToProps } from '../../../components/business-profile/BusinessProfile';

const { business, reviews, review, paginationResult} = businessData;
const { userUpdateData } = userData;
configure({ adapter: new Adapter() });

let props, store, match;
const setup = () => {
  props = {
    onChange: jest.fn(() => Promise.resolve()),
    reviewRequest: jest.fn(() => Promise.resolve()),
    deleteBusiness: jest.fn(() => Promise.resolve()),
    businessImageUploader: jest.fn(() => Promise.resolve()),
    uploadToCloudinary: jest.fn(() => Promise.resolve()),
    onStarClick: jest.fn(() => Promise.resolve()),
    onSubmit: jest.fn(() => Promise.resolve()),
    fetchReviews: jest.fn(() => Promise.resolve()),
    deleteReview: jest.fn(() => Promise.resolve()),
    deleteStatus: jest.fn(() => Promise.resolve()),
    onReviewDelete: jest.fn(() => Promise.resolve()),
    onReviewUpdate: jest.fn(() => Promise.resolve()),
    onReviewPageChange: jest.fn(() => Promise.resolve()),
    fetchBusinessById: jest.fn(() => Promise.resolve()),
    reviewUpdateRequest: jest.fn(() => Promise.resolve()),
    edit: jest.fn(() => Promise.resolve()),
    cancelEdit: jest.fn(() => Promise.resolve()),
    edit: jest.fn(() => Promise.resolve()),
    state: review,
    user: userUpdateData,
    match: {
      params: {
        id: 1
      }
    },
    params: {
      id: 1
    },
    reviews,
    paginate: paginationResult,
    loader: jest.fn(() => Promise.resolve()),
    business,
    isLoading: false,
    image: {image:'path/to/image'},
    history: {}
  };

  return shallow(<BusinessProfile {...props} />);
};


describe('Component: BusinessProfile', () => {
  beforeEach(() => {
    global.document = {
      title: () => {}
    },
    global.props = {
      match: {
        params: () => {}
      }
    }
    global.match = {
      params: () => {}
    }
  })

  it('should render', () => {
    const wrapper = setup();
    expect(wrapper.find('div').length).toBe(13);
    expect(wrapper.find('h4').length).toBe(3);
    expect(wrapper.find('p').length).toBe(4);
    expect(wrapper.find('span').length).toBe(4);
    expect(wrapper.find('ReactStars').length).toBe(1);
    expect(wrapper.find('Link').length).toBe(1);
    expect(wrapper.find('ImageZoom').length).toBe(1);
    expect(wrapper.find('BusinessImageUpload').length).toBe(1);
    expect(wrapper.length).toBe(1);
    // expect(wrapper.find('span').length).toBe(1);
    expect(wrapper.find('ReviewModal').length).toBe(1);
    expect(wrapper.find('ReviewList').length).toBe(0);
    wrapper.setProps({ user: {}, reviews: ['good'] });
    expect(wrapper.find('ReviewList').length).toBe(0);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});

describe('Component: BusinessProfile ComponentDidMount', () => {
  it('should get a business', () => {
    const spy = sinon.spy(BusinessProfile.prototype, 'componentDidMount');
    shallow(<BusinessProfile {...props} componentDidMount={spy}/>)
      .instance().componentDidMount({ setState: () => 1 });
  });
});

describe('Component: BusinessProfile ComponentDidMount', () => {
  it('should render connected component', () => {
    const wrapper = shallow(<connectedBusinessProfile store={store} />);
    expect(wrapper.length).toBe(1);
  }) 
});

describe('Component: BusinessProfile mapStateToProps', () => {
  beforeEach(() => {
    global.props = {
      match: {
        params: () => {}
      }
    }
    global.match = {
      match: () => {}
    }
  })

  // it('should check mapStateToProps', () => {
  //   const state = {
  //     businesses: {
  //       reviews,
  //       paginationResult
  //     },
  //     auth: userUpdateData,
  //     isLoading: false,
  //     image: ''
  //   };
  //   expect(mapStateToProps(state).businesses).toEqual(state);
  // }) 
});