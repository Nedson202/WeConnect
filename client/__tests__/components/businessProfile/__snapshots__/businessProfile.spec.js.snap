// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: BusinessProfile should render 1`] = `
<div>
  <div
    className="business-profile container"
  >
    <div
      className="row"
    >
      <div
        className="row col-sm-12 col-lg-10 offset-lg-1 profile-icon-style"
      >
        <div
          className="image-button-holder col-lg-4"
        >
          <ImageZoom
            defaultStyles={
              Object {
                "image": Object {},
                "overlay": Object {},
                "zoomContainer": Object {},
                "zoomImage": Object {},
              }
            }
            image={
              Object {
                "alt": "business image",
                "className": "img",
                "src": "image/connect.jpg",
                "style": Object {
                  "height": "300px",
                  "width": "320px",
                },
              }
            }
            onUnzoom={[Function]}
            onZoom={[Function]}
            shouldHandleZoom={[Function]}
            shouldReplaceImage={true}
            shouldRespectMaxDimension={false}
            zoomImage={
              Object {
                "alt": "business image",
                "src": "image/connect.jpg",
              }
            }
            zoomMargin={40}
          />
          <span>
            <button
              className="btn image-edit-button"
              data-target="#businessImageModal"
              data-toggle="modal"
            >
              <i
                className="fa fa-edit fa-lg"
              />
            </button>
          </span>
          <BusinessImageUpload
            onImageChange={[Function]}
            onImageSubmit={[Function]}
            params={
              Object {
                "id": 1,
              }
            }
            state={
              Object {
                "address": "2r close sau paulo",
                "averageRating": undefined,
                "category": undefined,
                "current": 1,
                "description": "this is a simple meal made with rice and yam",
                "editMessage": "",
                "editRating": 0,
                "editStatus": null,
                "email": "altech@caria.net",
                "errors": Object {},
                "image": "image/connect.jpg",
                "isLoading": false,
                "location": "sau paulo",
                "message": "",
                "name": "alpha tech",
                "rating": 0,
              }
            }
            uploadToCloudinary={[MockFunction]}
          />
        </div>
        <div
          className="col-lg-6 business-details"
        >
          <h4
            className="business-name text-capitalize"
          >
            alpha tech
          </h4>
          <h4>
            Description:
          </h4>
          <p>
            this is a simple meal made with rice and yam
          </p>
          <p>
            <i
              className="fa fa-envelope"
            />
            <span>
              altech@caria.net
            </span>
          </p>
          <p>
            <i
              className="fa fa-map-marker"
            />
            <span>
              2r close sau paulo sau paulo
            </span>
          </p>
          <p>
            <i
              className="fa fa-tags"
            />
            <span />
          </p>
          <ReactStars
            char="â˜…"
            color1="#333333"
            color2="#ffaf00"
            count={5}
            edit={false}
            half={true}
            onChange={[Function]}
            size={17}
            value={0}
          />
          <ul
            className="list-unstyled list-inline pull-right"
            style={
              Object {
                "paddingTop": "30px",
              }
            }
          >
            <li
              className="list-inline-item"
            >
              <Link
                className="btn edit-button"
                id="permission-button"
                replace={false}
                to="/updatebusiness/1"
              >
                <i
                  className="fa fa-edit fa-lg"
                />
                 edit
              </Link>
            </li>
            <li
              className="list-inline-item"
            >
              <button
                className="btn btn-danger"
                data-target="#businessDeleteModal"
                data-toggle="modal"
              >
                <i
                  className="fa fa-trash fa-lg"
                />
                 delete
              </button>
            </li>
          </ul>
          <ul
            className="list-unstyled list-inline pull-right"
            style={
              Object {
                "paddingBottom": "20px",
                "paddingTop": "30px",
              }
            }
          >
            <li
              className="list-inline-item"
            >
              <button
                className="btn btn-danger"
                data-target="#businessDeleteModal"
                data-toggle="modal"
              >
                <i
                  className="fa fa-trash fa-lg"
                />
                 delete
              </button>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className="col-sm-12 col-lg-10 offset-lg-1"
    >
      <ReviewModal
        business={
          Object {
            "address": "2r close sau paulo",
            "categories": "tech",
            "changeValue": 11,
            "color": "red",
            "defaultTotal": 250,
            "description": "this is a simple meal made with rice and yam",
            "email": "altech@caria.net",
            "id": 1,
            "image": "image/connect.jpg",
            "location": "sau paulo",
            "name": "alpha tech",
          }
        }
        isPosting={false}
        onChange={[Function]}
        onStarClick={[Function]}
        onSubmit={[Function]}
        state={
          Object {
            "address": "2r close sau paulo",
            "averageRating": undefined,
            "category": undefined,
            "current": 1,
            "description": "this is a simple meal made with rice and yam",
            "editMessage": "",
            "editRating": 0,
            "editStatus": null,
            "email": "altech@caria.net",
            "errors": Object {},
            "image": "image/connect.jpg",
            "isLoading": false,
            "location": "sau paulo",
            "message": "",
            "name": "alpha tech",
            "rating": 0,
          }
        }
        user={Object {}}
      />
    </div>
    <div
      className="col-sm-12 col-lg-10 offset-lg-1"
    >
      <ReviewsList
        cancelEdit={[Function]}
        deleteReview={[MockFunction]}
        deleteStatus={[Function]}
        edit={[Function]}
        fetchBusinessById={
          [MockFunction] {
            "calls": Array [
              Array [
                1,
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": Promise {},
              },
            ],
          }
        }
        fetchReviews={
          [MockFunction] {
            "calls": Array [
              Array [
                1,
                "page=1",
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": Promise {},
              },
            ],
          }
        }
        isLoading={false}
        onChange={[Function]}
        onReviewDelete={[Function]}
        onReviewPageChange={[Function]}
        onReviewUpdate={[Function]}
        onStarClick={[Function]}
        onSubmit={[Function]}
        paginate={
          Object {
            "limit": 6,
            "page": 1,
            "queryData": Object {
              "option": "option",
            },
            "totalBusiness": 10,
            "totalReviews": 10,
          }
        }
        params={
          Object {
            "id": 1,
          }
        }
        reviewUpdateRequest={[MockFunction]}
        reviews={
          Array [
            "good",
          ]
        }
        state={
          Object {
            "address": "2r close sau paulo",
            "averageRating": undefined,
            "category": undefined,
            "current": 1,
            "description": "this is a simple meal made with rice and yam",
            "editMessage": "",
            "editRating": 0,
            "editStatus": null,
            "email": "altech@caria.net",
            "errors": Object {},
            "image": "image/connect.jpg",
            "isLoading": false,
            "location": "sau paulo",
            "message": "",
            "name": "alpha tech",
            "rating": 0,
          }
        }
        user={Object {}}
      />
    </div>
  </div>
  <div
    aria-hidden="true"
    aria-labelledby="confirmModalTitle"
    className="modal fade"
    id="businessDeleteModal"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="modal-dialog modal-dialog-centered"
      role="document"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-body"
        >
          <h4>
            Are you sure?
          </h4>
        </div>
        <div
          className="modal-footer"
        >
          <button
            className="btn btn-outline-success"
            data-dismiss="modal"
            type="button"
          >
            No
          </button>
          <button
            className="btn btn-outline-success"
            data-dismiss="modal"
            onClick={[Function]}
            type="button"
          >
            Yes
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
